venv/Lib/site-packages/pip/_internal/network/lazy_wheel.py

```
@@ -0,0 +1,210 @@
+    during initialization.
+    """
+
+    def __init__(
+        self, url: str, session: PipSession, chunk_size: int = CONTENT_CHUNK_SIZE
+    ) -> None:
+        head = session.head(url, headers=HEADERS)
+        raise_for_status(head)
+        assert head.status_code == 200
+        self._session, self._url, self._chunk_size = session, url, chunk_size
+        self._length = int(head.headers["Content-Length"])
+        self._file = NamedTemporaryFile()
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)